docs/source/features/ray.rst

```diff
@@ -0,0 +1,377 @@
+
+Both resource-wrapped and tuning aggregate jobs dispatch individual jobs to a designated Ray
+cluster, which leverages the cluster's resources (e.g., a single workstation node or multiple nodes)
+to execute these jobs with workers in parallel and/or sequentially. By default, aggregate jobs use all \
+available resources on each available GPU-enabled node for each sub-job worker. This can be changed through
+specifying the ``--num_workers`` argument, especially critical for parallel aggregate
+job processing on local or virtual multi-GPU machines
+
+In resource-wrapped aggregate jobs, each sub-job and its
+resource requirements are defined manually, enabling resource isolation.
+For tuning aggregate jobs, individual jobs are generated automatically based on a hyperparameter
+sweep configuration. This assumes homogeneous node resource composition for nodes with GPUs.
+
+.. dropdown:: source/standalone/workflows/ray/wrap_resources.py
+  :icon: code
+
+  .. literalinclude:: ../../../source/standalone/workflows/ray/wrap_resources.py
+    :language: python
+    :emphasize-lines: 14-66
+
+.. dropdown:: source/standalone/workflows/ray/tuner.py
+  :icon: code
+
+  .. literalinclude:: ../../../source/standalone/workflows/ray/tuner.py
+    :language: python
+    :emphasize-lines: 18-53
+
+
+The following script can be used to submit aggregate
+jobs to one or more Ray cluster(s), which can be used for
+running jobs on a remote cluster or simultaneous jobs with heterogeneous
+resource requirements:
+
+.. dropdown:: source/standalone/workflows/ray/submit_job.py
+  :icon: code
+
+  .. literalinclude:: ../../../source/standalone/workflows/ray/submit_job.py
+    :language: python
+    :emphasize-lines: 12-53
+
+The following script can be used to extract KubeRay Cluster information for aggregate job submission.
+
+.. dropdown:: source/standalone/workflows/ray/grok_cluster_with_kubectl.py
+  :icon: code
+
+  .. literalinclude:: ../../../source/standalone/workflows/ray/grok_cluster_with_kubectl.py
+    :language: python
+    :emphasize-lines: 14-26
+
+The following script can be used to easily create clusters on Google GKE.
+
+.. dropdown:: source/standalone/workflows/ray/launch.py
+  :icon: code
+
+  .. literalinclude:: ../../../source/standalone/workflows/ray/launch.py
+    :language: python
+    :emphasize-lines: 16-37
+
+**Installation**
+----------------
+
+The Ray functionality requires additional dependencies be installed.
+
+To use Ray without Kubernetes, like on a local computer or VM,
+``kubectl`` is not required. For use on Kubernetes clusters with KubeRay,
+such as Google Kubernetes Engine or Amazon Elastic Kubernetes Service, ``kubectl`` is required, and can
+be installed via the `Kubernetes website <https://kubernetes.io/docs/tasks/tools/>`_
+
+The pythonic dependencies can be installed with:
+
+.. code-block:: bash
+
+  ./isaaclab.sh -p -m pip install 'ray[default, tune]==2.31.0'
```

Comment:
would it make sense to wrap all of these into `./isaaclab.sh -i ray`?

Reply:
I think that since this functionality is largely experimental, and requires different installations for different aspects of the functionality, it's ok to leave this in the docs for now. When/if it is more supported, then I think it would make sense to add it into the -i options